### ISSUE TITLE
Homework 5 Issue- Joining data files 

### ISSUE THREAD

**Issue by taylorburtch** [edquant/edh7916#18]
Lee and I are working on assignment 5 and are already stuck before question 1. We have entered the following:

```R
## read in data (then pipes) and lower all names using rename_all(tolower)
df_1 <- read_csv(file.path(dat_dir, "hd2007.csv")) %>%
    rename_all(tolower)

df_2 <- read_csv(file.path(dat_dir, "ic2007mission.csv")) 

df_joined <- df %>%
    left_join(df_1, df_2, by = "unitid")
```

We read in both data files and are stuck trying to join them by unitid, even after putting hd2007.csv in all lowercase. It is telling us 

```R
`by` can't contain join column `unitid` which is missing from LHS
```
Has anyone figured out how to begin this hw?

**Comment by taylorburtch** [comment]
I started to thinking maybe I was doing the join wrong, so I changed it to this... but I still don't know if this is right. 

```Rdf <- read_csv(file.path(dat_dir, "hd2007.csv")) %>%
    rename_all(tolower)

df_1 <- read_csv(file.path(dat_dir, "ic2007mission.csv")) 

df_joined <- df %>%
    left_join (df_1, by="unitid")
```
It seems like it will _let_ me do it now, but it does not look like df_joined has as many rows/columns as it should. I am not sure...

**Comment by btskinner** [comment]
@taylorburtch: in your two posts, where is `df` coming from? I only see that you've read in the two files as `df_1` and `df_2`.

Also, remember that if you if you use a pipe, then the output of the left goes into the input of the right. So, when you use 

```r
df_joined <- df %>%
    left_join(df_1, df_2, by = "unitid")
```
it is equivalent to typing
```r
df_joined <- left_join(df, df_1, df_2, by = "unitid")
```
which shouldn't work since `left_join()` can only join two data frames at a time.

**Comment by taylorburtch** [comment]
I actually changed it to 
```Rdf <- read_csv(file.path(dat_dir, "hd2007.csv")) %>%
    rename_all(tolower)

df_1 <- read_csv(file.path(dat_dir, "ic2007mission.csv")) 

df_joined <- df %>%
    left_join (df_1, by="unitid")

df_joined
```
Does this look more correct? 

I am stuck on parts of question 2 and 3. 
For question 2, I cannot figure out how I would control for punctuation in the "phd" pattern. So far I have 
```Rdf %>%
    distinct(chfnm) %>%
    filter(str_detect(str_to_lower(chfnm), "phd")) %>%
    select(ends_with("phd")) %>%
    count()
```
I don't know if I am even on the right track. Lee and I also could NOT figure out how to pull patterns from two columns for question 3. We thought maybe it would require the "str_subset" command from the cheat sheet... but I don't think we've used this in class yet. Has anyone else figured out how to do this??

**Comment by taylorburtch** [comment]
Something is not copying and pasting correct. It is only showing part of what I am entering for the first part under this issue page. Trying again. 

```Rdf <- read_csv(file.path(dat_dir, "hd2007.csv")) %>%
    rename_all(tolower)

df_1 <- read_csv(file.path(dat_dir, "ic2007mission.csv")) 

df_joined <- df %>%
    left_join (df_1, by="unitid")
```

**Comment by taylorburtch** [comment]
URG. Let me try putting it NOT in R text. 

df <- read_csv(file.path(dat_dir, "hd2007.csv")) %>%
    rename_all(tolower)

df_1 <- read_csv(file.path(dat_dir, "ic2007mission.csv")) 

df_joined <- df %>%
    left_join (df_1, by="unitid")

**Comment by inmanjes** [comment]
Are you still having trouble joining the data files?  I renamed everything to lower in both files and joined from there and it seemed to work.  Maybe try `rename_all(tolower)` on the second file?

**Comment by btskinner** [comment]
@taylorburtch: I've edited your second to last comment. I think the issue is that your first copied line was pasted to `r` just after the three back ticks. Make sure your first code line you want to show starts on its own line. Compare your [post I left alone](https://github.com/edquant/edh7916/issues/18#issuecomment-586740673) with the [one I fixed](https://github.com/edquant/edh7916/issues/18#issuecomment-586740777) by clicking the three dots and then `edit`. You'll see the raw code and what I mean.

I've also opened a new issue for your question about the bonus question: #19. Use that thread so we can keep things distinct.
